src/twisted/internet/endpoints.py

```diff
@@ -694,26 +717,43 @@ def connect(self, protocolFactory):
         connection which is established first.
         """
         wf = protocolFactory
-        d = self._nameResolution(self._host, self._port)
+        d = Deferred()
+        if self._badHostname:
+            ude = ValueError("invalid hostname: {}".format(self._hostText))
+            d.errback(ude)
+            return d
+        addresses = []
+        @provider(IResolutionReceiver)
+        class MyReceiver(object):
+            @staticmethod
+            def resolutionBegan(resolutionInProgress):
+                pass
+            @staticmethod
+            def addressResolved(address):
+                addresses.append(address)
+            @staticmethod
+            def resolutionComplete():
+                d.callback(addresses)
```

Comment:
I am not sure why TwistedChecker isn't complaining about the lack of newlines here :(